quetz/main.py

```diff
@@ -744,15 +744,16 @@ def delete_package_version(
 
 
 @api_router.get(
-    "/search/{query}", response_model=List[rest_models.PackageSearch], tags=["search"]
+    "/search/", response_model=List[rest_models.PackageSearch], tags=["search"]
 )
 def search(
-    query: str,
+    q: str,
     dao: Dao = Depends(get_dao),
     auth: authorization.Rules = Depends(get_rules),
 ):
     user_id = auth.get_user()
-    return dao.search_packages(query, user_id)
+    keywords, filters = parse_query(q)
+    return dao.search_packages(keywords, filters, user_id)
```

Comment:
we should also make sure that the user has permission to see the packages from the search results (based on channel and package permission)